<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>frameAllGenes — frameAllGenes • consexpressionR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="frameAllGenes — frameAllGenes"><meta name="description" content="This function creates a data.frame for each result of the diffrential expression method performed. It was created to
ensure that all genes been listed, even thought the gene does not appear as diffrentially expressed, in this case NA data is inserted.
This functis is executed into runExpression() function, so there is no need to call it separatelly."><meta property="og:description" content="This function creates a data.frame for each result of the diffrential expression method performed. It was created to
ensure that all genes been listed, even thought the gene does not appear as diffrentially expressed, in this case NA data is inserted.
This functis is executed into runExpression() function, so there is no need to call it separatelly."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">consexpressionR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/consexpressionR-vignette.html">consexpressionR-vignette</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>frameAllGenes</h1>

      <div class="d-none name"><code>frameAllGenes.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function creates a data.frame for each result of the diffrential expression method performed. It was created to
ensure that all genes been listed, even thought the gene does not appear as diffrentially expressed, in this case NA data is inserted.
This functis is executed into runExpression() function, so there is no need to call it separatelly.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">frameAllGenes</span><span class="op">(</span><span class="va">cons_list</span>, <span class="va">countMatrix</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-cons-list">cons_list<a class="anchor" aria-label="anchor" href="#arg-cons-list"></a></dt>
<dd><p>list generated by runExpression() function execution</p></dd>


<dt id="arg-countmatrix">countMatrix<a class="anchor" aria-label="anchor" href="#arg-countmatrix"></a></dt>
<dd><p>RData object with table count, where line name is gene name, and name column is treat name</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>list with analysis expression containing: all methods executed and their results, ensuring thar all genes has been listed;</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gse95077</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">treats</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BM"</span>, <span class="st">"JJ"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cons_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="runExpression.html">runExpression</a></span><span class="op">(</span>numberReplics <span class="op">=</span> <span class="fl">3</span>,</span></span>
<span class="r-in"><span>                             groupName <span class="op">=</span> <span class="va">treats</span>,</span></span>
<span class="r-in"><span>                              rDataFrameCount <span class="op">=</span> <span class="va">gse95077</span>,</span></span>
<span class="r-in"><span>                              sepCharacter <span class="op">=</span> <span class="st">","</span>,</span></span>
<span class="r-in"><span>                              experimentName <span class="op">=</span> <span class="st">"test_cons"</span>,</span></span>
<span class="r-in"><span>                              controlDeseq2 <span class="op">=</span> <span class="st">"BM"</span>,</span></span>
<span class="r-in"><span>                              contrastDeseq2 <span class="op">=</span> <span class="st">"JJ"</span>,</span></span>
<span class="r-in"><span>                              outDirPath <span class="op">=</span> <span class="st">"."</span> <span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using classic mode.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting annotation of the Homo Sapiens...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Using reference genome 38.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating gene expression values...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RQ fit ......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SQN </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  ===== ERROR: KnowSeq execution is failed === </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Error in mclustBIC(data = structure(c(-3.52446659876917, -2.42801002792396, : could not find function "mclustBIC"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ limma executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Computing (M,D) values..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Computing probability of differential expression..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ NOISeq executed! </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initial number of DE patterns = 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Final number of DE patterns = 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ EBSeq executed!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>some variables in design formula are characters, converting to factors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Dataset is COUNT data"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ DESeq2 executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating sequencing depths...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Resampling to get new data matrices...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 33</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 43</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 45</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 46</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 48</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 51</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 52</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 53</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 62</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 63</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 64</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 66</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 67</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 68</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 70</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 71</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 72</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 73</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 74</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 77</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 78</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 79</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 80</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 82</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 83</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 84</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 88</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 89</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 92</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 93</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 94</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 95</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 98</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 99</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of thresholds chosen (all possible thresholds) = 8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting all the cutoffs for the thresholds...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting number of false positives in the permutation...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ SAMSeq executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ DIFERENTIAL EXPRESSION ANALYSIS WAS COMPLETE!</span>
<span class="r-in"><span><span class="va">cons_rseult2</span> <span class="op">&lt;-</span> <span class="fu">frameAllGenes</span><span class="op">(</span>cons_list <span class="op">=</span> <span class="va">cons_result</span>,</span></span>
<span class="r-in"><span>                              countMatrix<span class="op">=</span><span class="va">gse95077</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  -  1 edger -  2 limma -  3 noiseq -  4 ebseq -  5 deseq2 -  6 samseq</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Juliana Costa-Silva, David Menotti, Fabricio M. Lopes.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer></div>





  </body></html>

