<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Make expression analysis of 7 tools, and return a list of results by each tool. — runExpression • consexpressionR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Make expression analysis of 7 tools, and return a list of results by each tool. — runExpression"><meta name="description" content="Make expression analysis of 7 tools, and return a list of results by each tool."><meta property="og:description" content="Make expression analysis of 7 tools, and return a list of results by each tool."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">consexpressionR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/consexpressionR-vignette.html">consexpressionR-vignette</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/costasilvati/consexpressionR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Make expression analysis of 7 tools, and return a list of results by each tool.</h1>
      <small class="dont-index">Source: <a href="https://github.com/costasilvati/consexpressionR/blob/master/R/runExpression.R" class="external-link"><code>R/runExpression.R</code></a></small>
      <div class="d-none name"><code>runExpression.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Make expression analysis of 7 tools, and return a list of results by each tool.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">runExpression</span><span class="op">(</span></span>
<span>  <span class="va">numberReplics</span>,</span>
<span>  <span class="va">groupName</span>,</span>
<span>  tableCountPath <span class="op">=</span> <span class="st">"data/gse95077.csv"</span>,</span>
<span>  sepCharacter <span class="op">=</span> <span class="st">","</span>,</span>
<span>  rDataFrameCount <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  experimentName <span class="op">=</span> <span class="st">"genericExperiment"</span>,</span>
<span>  outDirPath <span class="op">=</span> <span class="st">"../../consexpression2_results/"</span>,</span>
<span>  printResults <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  fitTypeDeseq2 <span class="op">=</span> <span class="st">"local"</span>,</span>
<span>  controlDeseq2 <span class="op">=</span> <span class="st">""</span>,</span>
<span>  contrastDeseq2 <span class="op">=</span> <span class="st">""</span>,</span>
<span>  methodNormLimma <span class="op">=</span> <span class="st">"TMM"</span>,</span>
<span>  methodAdjPvalueLimma <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  numberTopTableLimma <span class="op">=</span> <span class="fl">1e+06</span>,</span>
<span>  filterIdKnowseq <span class="op">=</span> <span class="st">"ensembl_gene_id"</span>,</span>
<span>  notSapiensKnowseq <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  methodNormEdgeR <span class="op">=</span> <span class="st">"TMM"</span>,</span>
<span>  normNoiseq <span class="op">=</span> <span class="st">"rpkm"</span>,</span>
<span>  kNoiseq <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  factorNoiseq <span class="op">=</span> <span class="st">"Tissue"</span>,</span>
<span>  lcNoiseq <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  replicatesNoiseq <span class="op">=</span> <span class="st">"technical"</span>,</span>
<span>  condExpNoiseq <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span>,</span>
<span>  respTypeSamseq <span class="op">=</span> <span class="st">"Two class unpaired"</span>,</span>
<span>  npermSamseq <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  fdrEbseq <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  maxRoundEbseq <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  methodDeResultsEbseq <span class="op">=</span> <span class="st">"robust"</span>,</span>
<span>  deNovoAanalysis <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  progressShiny <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-numberreplics">numberReplics<a class="anchor" aria-label="anchor" href="#arg-numberreplics"></a></dt>
<dd><p>number of replicate (technical or biological) by sample</p></dd>


<dt id="arg-groupname">groupName<a class="anchor" aria-label="anchor" href="#arg-groupname"></a></dt>
<dd><p>text, name of samples or treatment</p></dd>


<dt id="arg-tablecountpath">tableCountPath<a class="anchor" aria-label="anchor" href="#arg-tablecountpath"></a></dt>
<dd><p>path to csv file that contains count data or abundance data (local)</p></dd>


<dt id="arg-sepcharacter">sepCharacter<a class="anchor" aria-label="anchor" href="#arg-sepcharacter"></a></dt>
<dd><p>character used to split csv data, can be comma or tab</p></dd>


<dt id="arg-rdataframecount">rDataFrameCount<a class="anchor" aria-label="anchor" href="#arg-rdataframecount"></a></dt>
<dd><p>RData object with table count, where line name is gene name, and name column is treat name</p></dd>


<dt id="arg-experimentname">experimentName<a class="anchor" aria-label="anchor" href="#arg-experimentname"></a></dt>
<dd><p>text, name of experiment</p></dd>


<dt id="arg-outdirpath">outDirPath<a class="anchor" aria-label="anchor" href="#arg-outdirpath"></a></dt>
<dd><p>path to write output, need be a directory (local)</p></dd>


<dt id="arg-printresults">printResults<a class="anchor" aria-label="anchor" href="#arg-printresults"></a></dt>
<dd><p>logical variable: TRUE print report by each tool, FALSE print only consensus result</p></dd>


<dt id="arg-fittypedeseq-">fitTypeDeseq2<a class="anchor" aria-label="anchor" href="#arg-fittypedeseq-"></a></dt>
<dd><p>either "parametric", "local", "mean", or "glmGamPoi" for the type of fitting of dispersions to the mean intensity</p></dd>


<dt id="arg-controldeseq-">controlDeseq2<a class="anchor" aria-label="anchor" href="#arg-controldeseq-"></a></dt>
<dd><p>group of samples that represents control in experiment, used by DESeq2; Default is "".</p></dd>


<dt id="arg-contrastdeseq-">contrastDeseq2<a class="anchor" aria-label="anchor" href="#arg-contrastdeseq-"></a></dt>
<dd><p>specific reference level that you need compare (this name should be in groupName List)</p></dd>


<dt id="arg-methodnormlimma">methodNormLimma<a class="anchor" aria-label="anchor" href="#arg-methodnormlimma"></a></dt>
<dd><p>normalization method to be used (limma)</p></dd>


<dt id="arg-methodadjpvaluelimma">methodAdjPvalueLimma<a class="anchor" aria-label="anchor" href="#arg-methodadjpvaluelimma"></a></dt>
<dd><p>correction method, a character string. Can be abbreviated (limma)</p></dd>


<dt id="arg-numbertoptablelimma">numberTopTableLimma<a class="anchor" aria-label="anchor" href="#arg-numbertoptablelimma"></a></dt>
<dd><p>maximum number of genes to list (limma)</p></dd>


<dt id="arg-filteridknowseq">filterIdKnowseq<a class="anchor" aria-label="anchor" href="#arg-filteridknowseq"></a></dt>
<dd><p>The attribute used as filter to return the rest of the attributes.</p></dd>


<dt id="arg-notsapiensknowseq">notSapiensKnowseq<a class="anchor" aria-label="anchor" href="#arg-notsapiensknowseq"></a></dt>
<dd><p>A boolean value that indicates if the user wants the human annotation or another annotation available in BiomaRt. The possible not human dataset can be consulted by calling the following function: biomaRt::listDatasets(useMart("ensembl")).</p></dd>


<dt id="arg-methodnormedger">methodNormEdgeR<a class="anchor" aria-label="anchor" href="#arg-methodnormedger"></a></dt>
<dd><p>normalization method to be used in edgeR::calcNormFactors(), default: "TMM"</p></dd>


<dt id="arg-normnoiseq">normNoiseq<a class="anchor" aria-label="anchor" href="#arg-normnoiseq"></a></dt>
<dd><p>Normalization method t can be one of "rpkm" (default), "uqua" (upper quartile), "tmm" (trimmed mean of M) or "n" (no normalization). Default: "rpkm".</p></dd>


<dt id="arg-knoiseq">kNoiseq<a class="anchor" aria-label="anchor" href="#arg-knoiseq"></a></dt>
<dd><p>Counts equal to 0 are replaced by k. By default, k = 0.5</p></dd>


<dt id="arg-factornoiseq">factorNoiseq<a class="anchor" aria-label="anchor" href="#arg-factornoiseq"></a></dt>
<dd><p>A string indicating the name of factor whose levels are the conditions to be compared.</p></dd>


<dt id="arg-lcnoiseq">lcNoiseq<a class="anchor" aria-label="anchor" href="#arg-lcnoiseq"></a></dt>
<dd><p>Length correction is done by dividing expression by length^lc. By default, lc = 0</p></dd>


<dt id="arg-replicatesnoiseq">replicatesNoiseq<a class="anchor" aria-label="anchor" href="#arg-replicatesnoiseq"></a></dt>
<dd><p>In this argument, the type of replicates to be used is defined: "technical", "biological" or "no" replicates. By default, "technical" replicates option is chosen.</p></dd>


<dt id="arg-condexpnoiseq">condExpNoiseq<a class="anchor" aria-label="anchor" href="#arg-condexpnoiseq"></a></dt>
<dd><p>A vector containing the two conditions to be compared by the differential expression algorithm (needed when the factor contains more than 2 different conditions).</p></dd>


<dt id="arg-resptypesamseq">respTypeSamseq<a class="anchor" aria-label="anchor" href="#arg-resptypesamseq"></a></dt>
<dd><p>Problem type: "Quantitative" for a continuous parameter; "Two class unpaired" for two classes with unpaired observations; "Survival" for censored survival outcome; "Multiclass": more than 2 groups; "Two class paired" for two classes with paired observations. Default: "Two class unpaired".</p></dd>


<dt id="arg-npermsamseq">npermSamseq<a class="anchor" aria-label="anchor" href="#arg-npermsamseq"></a></dt>
<dd><p>Number of permutations used to estimate false discovery rates. Default 100</p></dd>


<dt id="arg-fdrebseq">fdrEbseq<a class="anchor" aria-label="anchor" href="#arg-fdrebseq"></a></dt>
<dd><p>parameter used in EBTest function: fdr False Discovery Rate cutt off</p></dd>


<dt id="arg-maxroundebseq">maxRoundEbseq<a class="anchor" aria-label="anchor" href="#arg-maxroundebseq"></a></dt>
<dd><p>parameter used in EBTest function: Number of iterations. The default value is 50.</p></dd>


<dt id="arg-methodderesultsebseq">methodDeResultsEbseq<a class="anchor" aria-label="anchor" href="#arg-methodderesultsebseq"></a></dt>
<dd><p>parameter used in GetDEResults function: "robust" or "classic". Using the "robust" option, EBSeq is more robust to genes with outliers and genes with extremely small variances. Using the "classic" option, the results will be more comparable to those obtained by using the GetPPMat() function from earlier version (&lt;= 1.7.0) of EBSeq</p></dd>


<dt id="arg-denovoaanalysis">deNovoAanalysis<a class="anchor" aria-label="anchor" href="#arg-denovoaanalysis"></a></dt>
<dd><p>boolean value (TRUE if dataset don`t have a reference genome)</p></dd>


<dt id="arg-progressshiny">progressShiny<a class="anchor" aria-label="anchor" href="#arg-progressshiny"></a></dt>
<dd><p>shiny app element, used to show execution progress</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>list with all analysis expression</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gse95077</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">treats</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BM"</span>, <span class="st">"JJ"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cons_result</span> <span class="op">&lt;-</span> <span class="fu">runExpression</span><span class="op">(</span>numberReplics <span class="op">=</span> <span class="fl">3</span>,</span></span>
<span class="r-in"><span>                              groupName <span class="op">=</span> <span class="va">treats</span>,</span></span>
<span class="r-in"><span>                              rDataFrameCount <span class="op">=</span> <span class="va">gse95077</span>,</span></span>
<span class="r-in"><span>                              sepCharacter <span class="op">=</span> <span class="st">","</span>,</span></span>
<span class="r-in"><span>                              experimentName <span class="op">=</span> <span class="st">"test_cons"</span>,</span></span>
<span class="r-in"><span>                              controlDeseq2 <span class="op">=</span> <span class="st">"BM"</span>,</span></span>
<span class="r-in"><span>                              contrastDeseq2 <span class="op">=</span> <span class="st">"JJ"</span>,</span></span>
<span class="r-in"><span>                              outDirPath <span class="op">=</span> <span class="st">"."</span> <span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using classic mode.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting annotation of the Homo Sapiens...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Using reference genome 38.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating gene expression values...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RQ fit ......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SQN </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  ===== ERROR: KnowSeq execution is failed === </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Error in mclustBIC(data = structure(c(-3.52446659876917, -2.42801002792396, : could not find function "mclustBIC"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ limma executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Computing (M,D) values..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Computing probability of differential expression..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ NOISeq executed! </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initial number of DE patterns = 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Final number of DE patterns = 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ EBSeq executed!</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>some variables in design formula are characters, converting to factors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Dataset is COUNT data"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ DESeq2 executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating sequencing depths...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Resampling to get new data matrices...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 33</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 43</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 45</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 46</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 48</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 51</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 52</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 53</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 62</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 63</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 64</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 66</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 67</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 68</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 70</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 71</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 72</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 73</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 74</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 77</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 78</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 79</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 80</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 82</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 83</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 84</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 88</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 89</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 92</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 93</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 94</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 95</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 98</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 99</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of thresholds chosen (all possible thresholds) = 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting all the cutoffs for the thresholds...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting number of false positives in the permutation...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ SAMSeq executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ DIFERENTIAL EXPRESSION ANALYSIS WAS COMPLETE!</span>
<span class="r-in"><span><span class="va">expDef_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="expressionDefinition.html">expressionDefinition</a></span><span class="op">(</span>resultTool <span class="op">=</span> <span class="va">cons_result</span>, groups <span class="op">=</span> <span class="va">treats</span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Juliana Costa-Silva, David Menotti, Fabricio M. Lopes.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer></div>





  </body></html>

