<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="listDeByTool: makes a data.frame with genes (lines) and all executed DE tool.
Each tool that identify gene as DE, are marked by 1."><title>listDeByTool: makes a data.frame with genes (lines) and all executed DE tool. Each tool that identify gene as DE, are marked by 1. — listDeByTool • consexpressionR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="listDeByTool: makes a data.frame with genes (lines) and all executed DE tool. Each tool that identify gene as DE, are marked by 1. — listDeByTool"><meta property="og:description" content="listDeByTool: makes a data.frame with genes (lines) and all executed DE tool.
Each tool that identify gene as DE, are marked by 1."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">consexpressionR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/consexpressionR-vignette.html">consexpressionR</a>
    <a class="dropdown-item" href="../articles/my-vignette.html">consexpressionR: User guide</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/costasilvati/consexpressionRdocs/articles/my-vignette.html">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>listDeByTool: makes a data.frame with genes (lines) and all executed DE tool. Each tool that identify gene as DE, are marked by 1.</h1>
      
      <div class="d-none name"><code>listDeByTool.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>listDeByTool: makes a data.frame with genes (lines) and all executed DE tool.
Each tool that identify gene as DE, are marked by 1.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">listDeByTool</span><span class="op">(</span><span class="va">consexpressionList</span>, <span class="va">deList</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>consexpressionList</dt>
<dd><p>A list generated by consexpressionR function</p></dd>


<dt>deList</dt>
<dd><p>List with names of all genes analysed in experiment</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>data.frame with genes (lines) and DE tools (column), where 1 shows a
DE indication by tool.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">cqn</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cons_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="runExpression.html">runExpression</a></span><span class="op">(</span>numberReplics <span class="op">=</span> <span class="fl">3</span>, groupName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BM"</span>, <span class="st">"JJ"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                              rDataFrameCount <span class="op">=</span> <span class="va">gse95077</span>,</span></span>
<span class="r-in"><span>                              sepCharacter <span class="op">=</span> <span class="st">","</span>,</span></span>
<span class="r-in"><span>                              experimentName <span class="op">=</span> <span class="st">"test_cons"</span>,</span></span>
<span class="r-in"><span>                              outDirPath <span class="op">=</span> <span class="st">"."</span> <span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using classic mode.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ edger executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting annotation of the Homo Sapiens...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Using reference genome 38.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating gene expression values...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RQ fit ......</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SQN </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The use of 'sig2' is deprecated; do specify 'sigma' (= sqrt(sig2)) instead</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Two classes detected, applying limma biclass</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ KnowSeq executed!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing intercept from test coefficients</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ limma executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Computing (M,D) values..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Computing probability of differential expression..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ NOISeq executed! </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>`expect_is()` was deprecated in the 3rd edition.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Use `expect_type()`, `expect_s3_class()`, or `expect_s4_class()` instead</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Initial number of DE patterns = 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Final number of DE patterns = 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ ebseq executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Dataset is COUNT data"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ DESeq2 executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Estimating sequencing depths...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Resampling to get new data matrices...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 33</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 37</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 43</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 45</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 46</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 48</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 51</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 52</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 53</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 57</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 58</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 62</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 63</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 64</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 65</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 66</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 67</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 68</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 70</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 71</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 72</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 73</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 74</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 77</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 78</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 79</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 80</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 81</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 82</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 83</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 84</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 85</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 88</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 89</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 91</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 92</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 93</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 94</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 95</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 96</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 98</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 99</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> perm= 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of thresholds chosen (all possible thresholds) = 8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting all the cutoffs for the thresholds...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Getting number of false positives in the permutation...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ SAMSeq executed!</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ------------ DIFERENTIAL EXPRESSION ANALYSIS WAS COMPLETE!</span>
<span class="r-in"><span><span class="va">expDef_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="expressionDefinition.html">expressionDefinition</a></span><span class="op">(</span>resultTool <span class="op">=</span> <span class="va">cons_result</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">deByTool</span> <span class="op">&lt;-</span> <span class="fu">listDeByTool</span><span class="op">(</span><span class="va">cons_result</span>, <span class="va">expDef_result</span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Juliana Costa-Silva, David Menotti, Fabricio M. Lopes.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

